lib/matplotlib/cm.py

```diff
@@ -46,6 +46,13 @@ def _gen_cmap_registry():
     for cmap in list(cmap_d.values()):
         rmap = cmap.reversed()
         cmap_d[rmap.name] = rmap
+
+    # Register colormap aliases for gray and grey.
+    cmap_d['grey'] = cmap_d['gist_grey'] = cmap_d['gray']
```

Comment:
Let's not make assumptions on equivalence and instead plainliy assign by the naming alias.
```suggestion
    cmap_d['grey'] = cmap_d['gray']
    cmap_d['gist_grey'] = cmap_d['gist_gray']
```
Please also move this section to before "Generate reversed cmaps" so that you do not need to handle the `*_r` cases yourself.